app/views/questionnaire.py

```diff
@@ -224,6 +235,48 @@ def get_thank_you(eq_id, form_type):  # pylint: disable=unused-argument
     return _redirect_to_latest_location(routing_path, collection_id, metadata.get('eq_id'), metadata.get('form_type'))
 
 
+@post_submission_blueprint.route('view-submission', methods=['GET'])
+@login_required
+def get_view_submission(eq_id, form_type):  # pylint: disable=unused-argument
+
+    session_data = get_session_store().session_data
+    submission_valid_until = int(dateutil.parser.parse(session_data.submitted_time).strftime('%s')) + session_data.submission_validity
```

Comment:
I think it would be clearer if we used `datetime.timedelta` and compared date objects rather than converting dates to ints. Something like:
```
submission_time = dateutil.parser.parse(session_data.submitted_time)
submission_valid_until = submission_time + datetime.timedelta(seconds=submission_validity)
```
The comparison would then be:
```
if submission_valid_until > datetime.utcnow():
```

This applies to the other places where dates are converted to ints